types/list.go

```diff
@@ -162,6 +162,9 @@ func (l List) Type(ctx context.Context) attr.Type {
 // ToTerraformValue returns the data contained in the AttributeValue as
 // a tftypes.Value.
 func (l List) ToTerraformValue(ctx context.Context) (tftypes.Value, error) {
+	if l.ElemType == nil {
+		return tftypes.Value{}, fmt.Errorf("cannot convert List to Terraform Value if ElemType field is null")
```

Comment:
Nit: I'd suggest being a little bit more explicit about the typing and using "missing" or "nil" over "null" (which is a Terraform specific concept in the type system)

```suggestion
		return tftypes.Value{}, fmt.Errorf("cannot convert List to tftypes.Value if ElemType field is not set")
```